### PR TITLE
Only traverse to store addr

### PR DESCRIPTION
**Pull request by mlevesquedion**
Investigating a new test case suggested by @PurelyApplied led me to find a related case where propagating to the `Value` in `Store` causes a spurious report to be produced:

```go

func TestStoringToTaintedAddrDoesNotTaintStoredValue() {
	myChan := make(chan string)
	s := core.Source{Data: "foo"}
	recv := <-myChan
	s.Data = recv
	core.Sink(recv)
	core.Sink(myChan)
}
```

On master, `core.Sink(recv)` in the above example yields a spurious report. (`core.Sink(myChan)` is there for additional certainty.)

Not propagating to the `Value` seems like the right thing to do, since it makes no sense. However, it does mean that we no longer handle this case:

```go
func TestPanicTypeAssertSource(i interface{}) {
	s := i.(core.Source)
	_ = s
	// The dominating type assertion would panic if i were not a source type.
	core.Sink(i) // TODO want "a source has reached a sink"
}
```

I think this is an acceptable regression, since this test was passing because we were doing a weird propagation, not because we were properly handling the semantics of the code.

Related issue: #188.

- [x] Tests pass
- [x] Running against a large codebase such as [Kubernetes](https://github.com/kubernetes/kubernetes) does not error out.
- (N/A) [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Review by PurelyApplied — APPROVED** [review]

**Comment by PurelyApplied** [comment]
> I think this is an acceptable regression, since this test was passing because we were doing a weird propagation, not because we were properly handling the semantics of the code.

I agree.  It does point out that we might want to distinguish why we are suppressing tests within the `// TODO` comments, though.  This test is suppressed because carrying information forward from a type assertion is currently out of scope.  The `// TODO want` is more of a "we might want to consider this case" than a "this is a regression."

tl;dr: we might want to adopt a habit of `// TODO(#<issue number>)`.

aaaaaand now I've taken too long to type this that you've already merged.  :)

**Comment by mlevesquedion** [comment]
Sorry about that.

I like the `TODO(#issue number)` idea a lot. I think we should have some kind of visibility into those as well, e.g. to make sure that:
1. We know about all the TODOs lurking in the code
2. Every TODO is associated with an issue
I'm thinking maybe something like a badge that shows the number of TODOs (we would want to drive this number down).
